bionic/deriver.py

```diff
@@ -150,52 +146,51 @@ def _get_ready_for_bootstrap_resolution(self):
         if self._is_ready_for_bootstrap_resolution:
             return
 
-        # Generate the static key spaces and tasks for each entity.
+        # Generate the static key spaces and tasks for each descriptor.
         self._key_spaces_by_dnode = {}
         self._task_lists_by_dnode = {}
         for name in self._flow_state.providers_by_name.keys():
             dnode = DescriptorNode.from_descriptor(name)
             self._populate_dnode_info(dnode)
 
-        # Create a state object for each task.
-        self._task_states_by_key = {}
+        # Create a lookup table for all tasks.
+        self._tasks_by_key = {}
         for tasks in self._task_lists_by_dnode.values():
             for task in tasks:
-                task_state = TaskState(task)
-                for key in task.keys:
-                    self._task_states_by_key[key] = task_state
+                for task_key in task.keys:
+                    self._tasks_by_key[task_key] = task
```

Comment:
We might have talked about this before but to be sure, would task list still be while task states can change with new descriptors?

Reply:
Yep, tasks will still be immutable.